### DISCUSSION TITLE
AKE in strongswan 6.0.0beta3

### DISCUSSION THREAD

**Discussion by yusuf-sami-acikgoz** [strongswan/strongswan#1823] [Q&A]
Hello, I am trying to multiple key exchange in IKEv2. I'm trying to use KE1_FRODO_AES_L1 for initiator and responder in proposals. The responder receives the proposal correctly, but on the responder I get the following error: "**an algorithm from private space would match, but peer implementation is unknown, skipped**". What could be causing the error?

I'm using the following plugins: **charon-systemd test-vectors random drbg nonce sha3 openssl pem pkcs1 revocation constraints pubkey frodo curl kernel-netlink socket-default updown vici**

**Comment by tobiasbrunner** [comment]
> What could be causing the error?

None of the PQC key exchange methods have standardized identifiers yet, they are currently allocated from the private use range (>= 1024).

So you have to enable `charon.send_vendor_id` on the initiator so it sends the strongSwan vendor ID that enables private identifiers on the server, or you could enable `charon.accept_private_algs` on the responder so it accepts any private identifiers even if no strongSwan vendor ID is received.
